### DISCUSSION TITLE
Does KGB algorithm support changing its objectives, constraints, and variables (with values) parameters at run time?

### DISCUSSION THREAD

**Discussion by muazhari** [anyoptimization/pymoo#622] [Q&A]
From what I know, KGB is a dynamic many objective optimization algorithm. It is expected to work in a dynamic environment, like changing objectives, constraints, and variables (and their values) at run time. So, the algorithm can adapt to give output based on the current input parameters. Does the KGB algorithm support it (yes/no)? And, can you provide a minimal example?

**Comment by blankjul** [comment]
KGB was contributed in #497. @bandundu can you help out and answer @muazhari questions?

The example is located here: https://github.com/anyoptimization/pymoo/blob/main/examples/algorithms/moo/kgb.py

**Comment by bandundu** [comment]
@muazhari - **Yes**, KGB supports changing objectives and changing variable values at runtime. However, the implementation does not support changing the amount of variables during runtime and is implemented only for unconstrained problems. It should also work with constraints and dynamic constraints but this is not available in the current implementation. This would be a valuable area for future research on this algorithm if you are interested ;)!
